include/deal.II/lac/solver.h

```diff
@@ -21,7 +21,10 @@
 #include <deal.II/lac/vector_memory.h>
 #include <deal.II/lac/solver_control.h>
 
+// Ignore deprecation warnings for auto_ptr.
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/signals2.hpp>
+DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
```

Comment:
Does this problem go away with a newer version of boost? In your case, do you use the bundled boost or an external one? If the latter, which version?